### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by infinisil**
https://editorconfig.org/

### CODE REVIEW

**Comment by infinisil** [comment]
We could also add CI to check that, but I don't think it's that important. I think it's just good practice for all repos to have one of those :)

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+indent_style = space
+
+[*.rs]
+indent_size = 4
```
Comment (philiptaron) [inline .editorconfig:11]: I like adding `max_line_length = 100` (maybe for all)

I would also like to use the nixfmt-rfc-style nix formatter!

https://github.com/rust-lang/rust/blob/master/.editorconfig

**Review by philiptaron — COMMENTED** [review]
What do you think?

File: .editorconfig (hunk 1 above)
Comment (infinisil) [reply .editorconfig:12]: Let's do Nix formatting with another PR, but the max line length sounds good, adding it for `[*]` :)

**Review by philiptaron — APPROVED** [review]

**Pull request merged**